service-base/src/main/java/org/eclipse/hono/service/http/HttpServiceBase.java

```diff
@@ -0,0 +1,461 @@
+     * <p>
+     *
+     * @return 8443
+     */
+    @Override
+    public int getInsecurePortDefaultValue() {
+        return 8080;
+    }
+
+    @Override
+    protected final int getActualPort() {
+        return (server != null ? server.actualPort() : Constants.PORT_UNCONFIGURED);
+    }
+
+    @Override
+    protected final int getActualInsecurePort() {
+        return (insecureServer != null ? insecureServer.actualPort() : Constants.PORT_UNCONFIGURED);
+    }
+
+    /**
+     * Invoked before the http server is started.
+     * <p>
+     * Subclasses may override this method to do any kind of initialization work.
+     *
+     * @return A future indicating the outcome of the operation. The start up process fails if the returned future
+     *         fails.
+     */
+    protected Future<Void> preStartServers() {
+        return Future.succeededFuture();
+    }
+
+    @Override
+    protected final Future<Void> startInternal() {
+
+        return preStartServers()
+                .compose(s -> checkPortConfiguration())
+                .compose(s -> {
+                    Router router = createRouter();
+                    if (router == null) {
+                        return Future.failedFuture("no router configured");
+                    } else {
+                        addRoutes(router);
+                        return CompositeFuture.all(bindSecureHttpServer(router), bindInsecureHttpServer(router));
+                    }
+                })
+                .compose(s -> onStartupSuccess());
+    }
+
+    /**
+     * Invoked after the http server has been started successfully.
+     * <p>
+     * May be overridden by sub-classes.
+     *
+     * @return A future indicating the outcome of the operation. The start up process fails if the returned future
+     *         fails.
+     */
+    protected Future<Void> onStartupSuccess() {
+
+        return Future.succeededFuture();
+    }
+
+    /**
+     * Creates the router for handling requests.
+     * <p>
+     * This method creates a router instance with the following routes:
+     * <ol>
+     * <li>A default route limiting the body size of requests to the maximum payload size set in the <em>config</em>
+     * properties.</li>
+     * <li>A route for retrieving this adapter's current status from the resource path returned by
+     * {@link #getStatusResourcePath()} (if not {@code null}).</li>
+     * </ol>
+     *
+     * @return The newly created router (never {@code null}).
+     */
+    protected Router createRouter() {
+
+        final Router router = Router.router(vertx);
+        LOG.info("limiting size of inbound request body to {} bytes", getConfig().getMaxPayloadSize());
+        router.route().handler(BodyHandler.create().setBodyLimit(getConfig().getMaxPayloadSize())
+                .setUploadsDirectory(DEFAULT_UPLOADS_DIRECTORY));
+
+        String statusResourcePath = getStatusResourcePath();
```

Comment:
I'd rather not add this route here and instead would rather get rid of the whole *status* resource as well. This should be done properly using Vert.x health checks.